xarray/core/dataarray.py

```diff
@@ -3430,6 +3432,225 @@ def pad(
+        else:
+            raise KeyError(f'Dimension "{dim}" does not have coordinates')
+
+        # These are dtypes with NaN values argmin and argmax can handle
+        na_dtypes = "cf0"
+
+        if skipna or (skipna is None and self.dtype.kind in na_dtypes):
+            # Need to skip NaN values since argmin and argmax can't handle them
+            allna = self.isnull().all(dim)
+            array = self.where(~allna, 0)
+            hasna = allna.any()
+        else:
+            array = self
+            allna = None
+            hasna = False
+
+        # If promote is None we only promote if there are NaN values.
+        if promote is None:
+            promote = hasna
+
+        if not promote and hasna and array.coords[dim].dtype.kind not in na_dtypes:
+            raise TypeError(
+                "NaN values present for NaN-incompatible dtype and Promote=False"
+            )
+
+        # This will run argmin or argmax.
+        indx = getattr(array, func)(
+            dim=dim, axis=None, keep_attrs=False, skipna=skipna, **kwargs
+        )
+
+        # Get the coordinate we want.
+        coordarray = array[dim]
+
+        # Handle dask arrays.
+        if isinstance(array, dask_array_type):
+            res = dask_array.map_blocks(coordarray, indx, dtype=indx.dtype)
+        else:
+            res = coordarray[
+                indx,
+            ]
+
+        # Promote to a dtype that can handle NaN values if needed.
+        newdtype, fill_value = dtypes.maybe_promote(res.dtype)
+        if promote and newdtype != res.dtype:
+            res = res.astype(newdtype)
+
+        # Put the NaN values back in after removing them, if necessary.
+        if hasna and allna is not None:
+            res = res.where(~allna, fill_value)
+
+        # The dim is gone but we need to remove the corresponding coordinate.
+        del res.coords[dim]
+
+        # Put the attrs back in if needed
+        if keep_attrs:
+            res.attrs = self.attrs
+
+        return res
+
+    def idxmin(
+        self,
+        dim: Optional[Hashable] = None,
+        axis: Optional[int] = None,
+        skipna: Optional[bool] = None,
+        promote: Optional[bool] = None,
+        keep_attrs: Optional[bool] = False,
+        **kwargs: Any,
+    ) -> "DataArray":
+        """Return the coordinate of the minimum value along a dimension.
+
+        Returns a new DataArray named after the dimension with the values of
+        the coordinate along that dimension corresponding to minimum value
+        along that dimension.
+
+        In comparison to `argmin`, this returns the coordinate while `argmin`
+        returns the index.
+
+        Parameters
+        ----------
+        dim : str (optional)
+            Dimension over which to apply `idxmin`.
+        axis : int (optional)
+            Axis(es) over which to repeatedly apply `idxmin`. Exactly one of
+            the 'dim' and 'axis' arguments must be supplied.
+        skipna : bool, optional
```

Comment:
```suggestion
        skipna : bool, default True
```

(and elsewhere, unless you / anyone has a strong view on the current state)

Reply:
Fixed, again the fix will come in another commit.